src/PatientChecking/PatientChecking/Controllers/PatientController.cs

```diff
@@ -132,5 +141,89 @@ public IActionResult Detail(int patientId)
 
             return View(model);
         }
+
+        [HttpPost]
+        public IActionResult Update(PatientDetailViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                ModelState.AddModelError("", "Unable to save changes. Please try again");
+            }
+
+            var patientDetails = new PatientDetails
+            {
+                PatientId = model.PatientId,
```

Comment:
Just a suggestion: In the future, we can use `AutoMapper` for mapping these model. The mapping profile can be reused for other places. We can keep this for now.

Reply:
Thanks for your suggestion. I will apply it in the future